src/hb-ot-name-table.hh

```diff
@@ -197,8 +197,6 @@ struct name
 
     if (unlikely (c->ran_out_of_room)) return_trace (false);
 
-    assert (this->stringOffset == c->length ());
```

Comment:
Why remove?  If overflow happens, we should detect it earlier and return.

Reply:
The line a few lines earlier where we have this:
```
   this->stringOffset = c->length ();
```
It should become something like:
```
  if (unlikely (!c->check_assign (this->stringOffset, c->length ())) return_trace (false);
```

Reply:
oh, so you don't check in sanitize() for invalid tables but rather do checks in serializer.

Reply:
sorry, I have a question here: 
we have a task to remove return type of bool from all serialize calls. Is serializer be able to capture this error state in this case?

Reply:
Yes, the `check_assign` thing records the failure.

Reply:
> oh, so you don't check in sanitize() for invalid tables but rather do checks in serializer.

The job of `sanitize` is only to make sure the rest of the methods to *access* the table don't cause invalid memory access.  We are not in the font law enforcement business :D.

Reply:
got it. Thanks! :)